### DISCUSSION TITLE
Block width error

### DISCUSSION THREAD

**Discussion by NovakEvana** [ORTools4MTEX/ORTools#20] [Q&A] [answered]
Hello Frank,

Could you take a look at the error below that was derived from the `plotMap_blockWidths`? I'm using the ORTools-2.0.0. Thanks Frank.

![image](https://user-images.githubusercontent.com/117882416/203464189-7dee0bf2-3b8d-437a-a699-7ff8399dae9d.png)

**Comment by frankNiessen** [comment, marked as answer]
Hi, 

could you give me some more background information to help me finding the bug? Can you maybe share the dataset with me?
I will try to find the issue, but it is impossible from your shared error message.

Best wishes
Frank

**Comment by NovakEvana** [comment]
Hello Frank,

I attached the dataset at the [link ](https://drive.google.com/drive/folders/184uYg5jN1IpZtqxWb4uolEnWMTLkIhsT?usp=sharing)here.
The sample code is based on the OR_Example_7. Please let me know if you have any questions.

Thanks Frank.

**Reply by frankNiessen** [reply]
Hi, thank you for sharing the code and data. I have now checked it and have indeed found a bug. While looking for the packet ID's of child grains within the parent grain, we did not account for that not all grains in a parent grain are necessarily of the child phase.

Please download the latest release and let me know if this fixed your problem. 
https://github.com/ORTools4MTEX/ORTools/releases/tag/v2.0.1

Best wishes
Frank

**Reply by NovakEvana** [reply]
Hello Frank, 
Thanks for working on that. The code for separating grain is working perfectly; however, the grain graph one `plotMap_blockWidths(job,'linewidth',1);` showed up with just a few grains. Could you take a look at this again? Thanks Frank.

P/S: I think it's connected with the problem posted [here ](https://github.com/ORTools4MTEX/ORTools/discussions/18#discussioncomment-3908965) but didn't see your response on this. 

![image](https://user-images.githubusercontent.com/117882416/204031735-16c84f0f-464e-43d6-9f6e-04bd4fd15c99.png)

**Reply by frankNiessen** [reply]
Is this on the same dataset you shared?

**Reply by NovakEvana** [reply]
Yes, Frank.

**Reply by frankNiessen** [reply]
Hi Evana, you are right, it is related to the [previous problem](https://github.com/ORTools4MTEX/ORTools/discussions/18#discussioncomment-3908965b). 
I am quite sure that it is fixed now, I made a new release again:

https://github.com/ORTools4MTEX/ORTools/releases/tag/v2.0.1

Thanks for helping me catch the bug, please let me know if it works.

Best wishes
Frank

**Reply by NovakEvana** [reply]
Hi Frank, I can't expect more from you. Thank you so much! It's working now. Just a minor ask, I think you haven't published the new version yet. I fixed my code based on your changes on the commitment page. 

Thanks Frank, have a nice day!

**Reply by frankNiessen** [reply]
Great, it works, I have made a new release 2.0.2 now.
